Source/Scene/ImplicitTileCoordinates.js

```diff
@@ -178,12 +219,322 @@ ImplicitTileCoordinates.prototype.deriveChildCoordinates = function (
+};
+
+/**
+ * Get the coordinates of the parent subtree that contains this tile
+ *
+ * @returns {ImplicitTileCoordinates} The parent subtree that contains this tile
+ * @private
+ */
+ImplicitTileCoordinates.prototype.deriveParentSubtreeCoordinates = function () {
+  var subtreeCoordinates = this.deriveSubtreeCoordinates();
+  if (subtreeCoordinates.isRoot()) {
+    // This subtree is the root of the entire tileset, so it can't go up any higher.
+    return subtreeCoordinates;
+  }
+
+  var subdivisionScheme = subtreeCoordinates.subdivisionScheme;
+  var subtreeLevels = subtreeCoordinates.subtreeLevels;
+  var subtreeLevel = subtreeCoordinates.level;
+  var subtreeX = subtreeCoordinates.x;
+  var subtreeY = subtreeCoordinates.y;
+
+  var parentSubtreeLevel = subtreeLevel - subtreeLevels;
+  var parentSubtreeX = subtreeX >> subtreeLevels;
+  var parentSubtreeY = subtreeY >> subtreeLevels;
+
+  if (subdivisionScheme === ImplicitSubdivisionScheme.OCTREE) {
+    var subtreeZ = subtreeCoordinates.z;
+    var parentSubtreeZ = subtreeZ >> subtreeLevels;
+
+    return new ImplicitTileCoordinates({
+      subdivisionScheme: subdivisionScheme,
+      subtreeLevels: subtreeLevels,
+      level: parentSubtreeLevel,
+      x: parentSubtreeX,
+      y: parentSubtreeY,
+      z: parentSubtreeZ,
+    });
+  }
+
+  // Quadtree
+  return new ImplicitTileCoordinates({
+    subdivisionScheme: subdivisionScheme,
+    subtreeLevels: subtreeLevels,
+    level: parentSubtreeLevel,
+    x: parentSubtreeX,
+    y: parentSubtreeY,
+  });
+};
+
+/**
+ * Get the local coordinates of the tile relative to its subtree
+ *
+ * @returns {ImplicitTileCoordinates} The local coordinates of the tile
+ * @private
+ */
+ImplicitTileCoordinates.prototype.deriveLocalTileCoordinates = function () {
+  var subdivisionScheme = this.subdivisionScheme;
+  var subtreeLevels = this.subtreeLevels;
+  var tileX = this.x;
+  var tileY = this.y;
+  var tileLevel = this.level;
+
+  var subtreeCoordinates = this.deriveSubtreeCoordinates();
+  var localLevel = tileLevel - subtreeCoordinates.level;
+
+  var localX = tileX % (1 << localLevel);
+  var localY = tileY % (1 << localLevel);
+
+  if (subdivisionScheme === ImplicitSubdivisionScheme.OCTREE) {
+    var tileZ = this.z;
+    var localZ = tileZ % (1 << localLevel);
+
+    return new ImplicitTileCoordinates({
+      subdivisionScheme: subdivisionScheme,
+      subtreeLevels: subtreeLevels,
+      level: localLevel,
+      x: localX,
+      y: localY,
+      z: localZ,
+    });
+  }
+
+  // Quadtree
+  return new ImplicitTileCoordinates({
+    subdivisionScheme: subdivisionScheme,
+    subtreeLevels: subtreeLevels,
+    level: localLevel,
+    x: localX,
+    y: localY,
+  });
+};
+
+/**
+ * Get the local coordinates of the child subtree relative to its subtree
+ *
+ * @returns {ImplicitTileCoordinates} The local coordinates of the child subtree
+ * @private
+ */
+ImplicitTileCoordinates.prototype.deriveLocalChildSubtreeCoordinates = function () {
```

Comment:
Also another thing I notice about this file is we have a lot of functions thrown around, but the symmetries are a bit unclear.

We might want to discuss this in more depth, maybe over a whiteboard to plan the most concise set of orthogonal functions here.